compile/api/src/main/scala/sbt/ClassToAPI.scala

```diff
@@ -67,21 +68,26 @@ object ClassToAPI {
     }
 
   /** Returns the (static structure, instance structure, inherited classes) for `c`. */
-  def structure(c: Class[_], enclPkg: Option[String], cmap: ClassMap): (api.Structure, api.Structure) =
-    {
-      val methods = mergeMap(c, c.getDeclaredMethods, c.getMethods, methodToDef(enclPkg))
-      val fields = mergeMap(c, c.getDeclaredFields, c.getFields, fieldToDef(enclPkg))
-      val constructors = mergeMap(c, c.getDeclaredConstructors, c.getConstructors, constructorToDef(enclPkg))
-      val classes = merge[Class[_]](c, c.getDeclaredClasses, c.getClasses, toDefinitions(cmap), (_: Seq[Class[_]]).partition(isStatic), _.getEnclosingClass != c)
-      val all = methods ++ fields ++ constructors ++ classes
-      val parentJavaTypes = allSuperTypes(c)
-      if (!Modifier.isPrivate(c.getModifiers))
-        cmap.inherited ++= parentJavaTypes.collect { case c: Class[_] => c }
-      val parentTypes = types(parentJavaTypes)
-      val instanceStructure = new api.Structure(lzyS(parentTypes.toArray), lzyS(all.declared.toArray), lzyS(all.inherited.toArray))
-      val staticStructure = new api.Structure(lzyEmptyTpeArray, lzyS(all.staticDeclared.toArray), lzyS(all.staticInherited.toArray))
-      (staticStructure, instanceStructure)
+  def structure(c: Class[_], enclPkg: Option[String], cmap: ClassMap): (api.Structure, api.Structure) = {
+    lazy val cf = {
+      // TODO: need better way to get access to classfile-location
```

Comment:
I think maybe the TODO shoudl be about reading as much of the APi as possible from a classfile parser rather than reflection.  Reflection locks us into a specific JDK we have to be running on, where as parsing frees that.